tests/parser/tst_parser.cpp

```diff
@@ -757,32 +757,32 @@ void tst_Parser::mapsAndArrays()
                    "{_ 1: [_ " + expected + "], \"Hello\": {_ " + expected + ": (_ )}}");
 }
 
-struct Input {
-    QByteArray data;
-    int consumed;
-};
-
 static const CborParserOperations byteArrayOps = {
     /* can_read_bytes = */ [](const CborValue *value, size_t len) {
-        auto input = static_cast<Input *>(value->parser->data.ctx);
-        return input->data.size() - input->consumed >= int(len);
+        auto data = static_cast<QByteArray *>(value->parser->data.ctx);
+        auto consumed = uintptr_t(value->source.token);
+        return uintptr_t(data->size()) - consumed >= uintptr_t(len);
     },
     /* read_bytes = */ [](const CborValue *value, void *dst, size_t offset, size_t len) {
-        auto input = static_cast<Input *>(value->parser->data.ctx);
-        return memcpy(dst, input->data.constData() + input->consumed + offset, len);
+        auto data = static_cast<QByteArray *>(value->parser->data.ctx);
+        auto consumed = uintptr_t(value->source.token);
+        return memcpy(dst, data->constData() + consumed + offset, len);
     },
     /* advance_bytes = */ [](CborValue *value, size_t len) {
-        auto input = static_cast<Input *>(value->parser->data.ctx);
-        input->consumed += int(len);
+        auto consumed = uintptr_t(value->source.token);
+        consumed += int(len);
+        value->source.token = (void*)consumed;
```

Comment:
Please write `reinterpret_cast<void *>(consumed)`

Reply:
Fixed, and also did the same for the cast done in `transfer_string` as well.